test/new-e2e/tests/fips-compliance/fips_nix_test.go

```diff
@@ -0,0 +1,77 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package fipscompliance
+
+import (
+	_ "embed"
+	"fmt"
+
+	"testing"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments"
+	awshost "github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments/aws/host"
+	"github.com/DataDog/test-infra-definitions/components/os"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/ec2"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+//go:embed fixtures/openssl-default.cnf
+var defaultOpenSSLConfig string
+
+type LinuxFIPSComplianceSuite struct {
+	e2e.BaseSuite[environments.Host]
+}
+
+func TestLinuxFIPSComplianceSuite(t *testing.T) {
+	e2e.Run(t, &LinuxFIPSComplianceSuite{},
+		e2e.WithProvisioner(awshost.ProvisionerNoFakeIntake(
+			awshost.WithEC2InstanceOptions(ec2.WithOS(os.UbuntuDefault)),
+		)))
+}
+
+func (v *LinuxFIPSComplianceSuite) TestFIPSDefaultConfig() {
+	status := v.Env().RemoteHost.MustExecute("sudo GOFIPS=0 datadog-agent status")
+	assert.NotContains(v.T(), status, "can't enable FIPS mode for OpenSSL")
+	assert.Contains(v.T(), status, "Status date")
+
+	status = v.Env().RemoteHost.MustExecute("sudo GOFIPS=1 datadog-agent status")
+	assert.NotContains(v.T(), status, "can't enable FIPS mode for OpenSSL")
+	assert.Contains(v.T(), status, "Status date")
+}
+
+func (v *LinuxFIPSComplianceSuite) TestFIPSNoFIPSProvider() {
+	v.Env().RemoteHost.MustExecute("sudo mv /opt/datadog-agent/embedded/ssl/openssl.cnf /opt/datadog-agent/embedded/ssl/openssl.cnf.tmp")
+	output := v.Env().RemoteHost.MustExecute(fmt.Sprintf(`sudo sh -c "echo '%s' > /opt/datadog-agent/embedded/ssl/openssl.cnf"`, defaultOpenSSLConfig))
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏You might want to use `v.Env().RemoteHost.WriteFile` (or other similar methods) to be safer, this way would fail if the config contained single quotes for example, and could even lead to executing unwanted commands

Reply:
2b616725733590d87a7c46b756dea1c8be63b523